tap_nice_incontact/streams.py

```diff
@@ -604,6 +612,188 @@ def get_records(self,
                         for rec in results.get(self.data_key))
 
 
+class DataExtractionStream(IncrementalStream):
+    """
+    Base class for streams pulling data via the Data Extraction API endpoints
+    Docs: https://help.nice-incontact.com/content/recording/dataextractionapi.htm
+    :param client: The API client used extract records from the external source
+    """
+    replication_method = 'INCREMENTAL'
+    entity_name = ''
+
+    # amount of time to delay before polling again
+    poll_delay = 5
+    # number of seconds before timing out
+    poll_timeout = 300
+    default_period = 'days'
+
+    path = 'jobs'
+
+    def get_status_uri(self, job_id):
+        return f'{self.path}/{job_id}'
+
+    def download_csv(self, csv_uri):
+        rows = []
+        with requests.get(csv_uri, stream=True) as r:
+            lines = (line.decode('utf-8') for line in r.iter_lines())
+            for row in csv.DictReader(lines):
+                rows.append(row)
+        return rows
+
+    def create_job(self, start_date, end_date):
+        """ Create a new job in the data extraction API """
+        if DEBUG_JOBS:
+            # Set a fixed job ID for debugging
+            return DEBUG_JOB_ID
```

Comment:
Can just check truthy on this

```suggestion
        if DEBUG_JOB_ID:
            # Set a fixed job ID for debugging
            return DEBUG_JOB_ID
```

Reply:
Just removing this so its cleaner to read